relayer/provider/cosmos/tx.go

```diff
@@ -1178,6 +1178,57 @@ func (cc *CosmosProvider) MsgTimeoutOnClose(ctx context.Context, msgRecvPacket p
 	return NewCosmosMessage(msgTimeout), nil
 }
 
+func (cc *CosmosProvider) MsgUpdateClient(latestHeader provider.IBCHeader, clientTrustedState provider.ClientTrustedState, signer string) (provider.RelayerMessage, error) {
+	var signedHeader *tmtypes.SignedHeader
+	var validatorSet, trustedValidators *tmtypes.ValidatorSet
+
+	switch trustedHeader := clientTrustedState.IBCHeader.(type) {
+	case CosmosIBCHeader:
+		trustedValidators = trustedHeader.ValidatorSet
+	default:
+		return nil, fmt.Errorf("Unsupported IBC trusted header type, only CosmosIBCHeader currently supported: %v", trustedHeader)
+	}
```

Comment:
Great suggestion! I split it up so that the header assembly is done by the counterparty, and the message assembly is done by the destination. This allowed the message assembly to consolidate into the existing `UpdateClient` provider method also.